internal/users/users_edit.go

```diff
@@ -0,0 +1,140 @@
+package users
+
+import (
+	"Twitter_like_application/internal/database/pg"
+	"Twitter_like_application/internal/services"
+	"encoding/json"
+	"fmt"
+	"golang.org/x/crypto/bcrypt"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func EditProfile(w http.ResponseWriter, r *http.Request) {
+	userID := r.Context().Value("userID").(int)
+	err := updateProfile(w, r, userID)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	response := map[string]interface{}{
+		"status":  "success",
+		"message": "Profile updated successfully",
+	}
+	w.Header().Set("Content-Type", "application/json")
+	json.NewEncoder(w).Encode(response)
+}
+func updateProfile(w http.ResponseWriter, r *http.Request, userID int) error {
+	nameRegex := regexp.MustCompile(`^[a-zA-Z]+$`)
+	var updatedProfile Users
+	err := json.NewDecoder(r.Body).Decode(&updatedProfile)
+	if err != nil {
+		return fmt.Errorf("failed to decode request body: %v", err)
+	}
+	values := []any{}
+	key := []string{}
+	if updatedProfile.Name != "" {
+		if !nameRegex.MatchString(updatedProfile.Name) {
+			services.ReturnErr(w, "Invalid name format", http.StatusBadRequest)
+			return err
+		}
+		if len(updatedProfile.Name) > 70 {
+			services.ReturnErr(w, "Name exceeds maximum length", http.StatusBadRequest)
+			return err
+		}
+		values = append(values, updatedProfile.Name)
+		key = append(key, " name = $"+strconv.Itoa(len(key)+1))
+	}
+	if updatedProfile.Password != "" {
+		if !nameRegex.MatchString(updatedProfile.Password) {
```

Comment:
this seems wrong, first, limiting password with alphabetical chars, second, there was another regexp for password in another place, there should be only one. Otherwise you risk getting different rules for the same field.